localization/pose_instability_detector/README.md

```diff
@@ -1,20 +1,89 @@
+Ideally, `pose_instability_detector` needs the twist values between the previous pose and the current pose.
+Therefore, `pose_instability_detector` snips the twist buffer and apply interpolations and extrapolations to obtain the twist values at the desired time.
+
+![how_to_snip_necessary_twist](./media/how_to_snip_twist.png)
+
+### Linear transition and angular transition
+
+After the twist values are collected, the node calculates the linear transition and angular transition based on the twist values and add them to the previous pose.
+
+## Threshold definition
 
-![timeline](./media/timeline.drawio.svg)
+The `pose_instability_detector` node compares the pose calculated by dead reckoning with the latest pose from the EKF output.
+These two pose are ideally the same, but in reality, they are not due to the error in the twist values the pose observation.
+If these two poses are significantly different so that the absolute value exceeds the threshold, the node outputs a WARN message to the `/diagnostics` topic.
+There are six thresholds (x, y, z, roll, pitch, and yaw) to determine whether the poses are significantly different, and these thresholds are determined by the following subsections.
+
+### `diff_position_x`
+
+This threshold examines the difference in the longitudinal axis between the two poses.
+
+$$
+\tau_x = v_{\rm max}\frac{\beta_v}{100} \Delta t + \epsilon_x
+$$
+
+- $\tau_x \cdots$ Threshold for the difference in the longitudinal axis
+- $v_{\rm max} \cdots$ Maximum velocity [m/s]
+- $\beta_v \cdots$ Scale factor tolerance for the maximum velocity [%]
+- $\Delta t \cdots$ Time interval [s]
+- $\epsilon_x \cdots$ Pose estimator (e. g. ndt_scan_matcher) error tolerance in the longitudinal axis [m]
+
+### `diff_position_y` and `diff_position_z`
+
+These thresholds examine the difference in the lateral and vertical axes between the two poses.
+The `pose_instaibility_detector` calculates the possible range where the vehicle goes, and get the maximum difference between the nominal dead reckoning pose and the maximum limit pose.
+
+![lateral_threshold_calculation](./media/lateral_threshold_calculation.png)
+
+### `diff_orientation_roll`, `diff_orientation_pitch`, and `diff_orientation_yaw`
+
+These thresholds examine the difference in the roll, pitch, and yaw angles between the two poses.
+
+$$
+\tau_\phi = \tau_\theta = \tau_\psi = \left(\omega_{\rm max}\frac{\beta_\omega}{100} + b \right) \Delta t + 3\sigma_\omega \sqrt{\Delta t} + \epsilon_\psi
+$$
+
+- $\tau_\phi, \tau_\theta, \tau_\psi \cdots$ Threshold for the difference in the roll, pitch, and yaw angles
+- $\omega_{\rm max} \cdots$ Maximum angular velocity [rad/s]
+- $\beta_\omega \cdots$ Scale factor tolerance for the maximum angular velocity [%]
+- $b \cdots$ Bias tolerance of the angular velocity [rad/s]
+- $\sigma_\omega \cdots$ Standard deviation of the angular velocity [rad/s]
```

Comment:
I think the unit of  $\sigma_\omega$ is [ $rad/s^{3/2}$ ] .  Is it right?

Because the above equation has $3\sigma_\omega \sqrt{\Delta t} [rad/s]$